src/evaluate/evaluator/base.py

```diff
@@ -218,22 +224,112 @@ def prepare_metric(self, metric: Union[str, EvaluationModule]):
+            data (`str` or `Dataset`, defaults to `None`):
+                Specifies the dataset we will run evaluation on. If it is of type `str`, we treat it as the dataset
+                name, and load it. Otherwise we assume it represents a pre-loaded dataset.
+            metric (`str` or `EvaluationModule`, defaults to `None`):
+                Specifies the metric we use in evaluator. If it is of type `str`, we treat it as the metric name, and
+                load it. Otherwise we assume it represents a pre-loaded metric.
+            preprocessor (`str` or `PreTrainedTokenizerBase` or `FeatureExtractionMixin`, *optional*, defaults to `None`):
+                Argument can be used to overwrite a default preprocessor if `model_or_pipeline` represents a model for
+                which we build a pipeline. If `model_or_pipeline` is `None` or a pre-initialized pipeline, we ignore
+                this argument.
+            strategy: (`Literal["simple", "bootstrap"]`, defaults to "simple"):
+                specifies the evaluation strategy. Possible values are:
+                - `"simple"` - we evaluate the metric and return the scores.
+                - `"bootstrap"` - on top of computing the metric scores, we calculate the confidence interval for each
+                of the returned metric keys, using `scipy`'s `bootstrap` method
+                https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bootstrap.html.
+            confidence_level (`float`, defaults to `0.95`):
+                The `confidence_level` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            n_resamples (`int`, defaults to `9999`):
+                The `n_resamples` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            random_state (`int`, *optional*, defaults to `None`):
+                The `random_state` value passed to `bootstrap` if `"bootstrap"` strategy is chosen. Useful for
+                debugging.
+            input_column (`str`, defaults to `"text"`):
+                the name of the column containing the text feature in the dataset specified by `data`.
+            label_column (`str`, defaults to `"label"`):
+                the name of the column containing the labels in the dataset specified by `data`.
+            label_mapping (`Dict[str, Number]`, *optional*, defaults to `None`):
+                We want to map class labels defined by the model in the pipeline to values consistent with those
+                defined in the `label_column` of the `data` dataset.
+        Return:
+            A `Dict`. The keys represent metric keys calculated for the `metric` spefied in function arguments. For the
+            `"simple"` strategy, the value is the metric score. For the `"bootstrap"` strategy, the value is a `Dict`
+            containing the score, the confidence interval and the standard error calculated for each metric key.
+        Examples:
```

Comment:
I am not sure about aggregating examples here, and actually the unified `compute()` documentation. For example, the default may be different depending on the subclass (`TextClassificationEvaluator`, `TokenClassificationEvaluator`). We may have different arguments for different subclass `compute()`, for example https://github.com/fxmarty/evaluate/blob/540956fb081cf909d29df02cb70dbab7db828b37/src/evaluate/evaluator/question_answering.py#L92-L93 .

Maybe we could do something like:

```python
@add_start_docstrings(COMPUTE_ARGS)
@add_start_docstrings(BOOTSTRAP_ARGS)
def compute(
    self,
    subclass_specific_arg1,
    subclass_specific_arg2,
    **kwargs,
) -> Tuple[Dict[str, float], Any]:
    """
    subclass-specific doc here
    """
    super().compute(...)  # or redefine compute here if necessary, passing subclass specific args to the parent's compute() is not defined
```

I feel like the parent's `compute()` will rapidly be irrelevant for specific subclasses if we add them, but I'm not sure. For question-answering it would not fit, I think.